src/app/board/board.component.ts

```diff
@@ -9,67 +9,54 @@ export interface Criterion {
   text: string
 }
 
-const questions = [
-  { text: "Quel temps fait-il aujourd'hui?" },
-  { text: "Comment vas-tu?" },
-  { text: "C'est bien Angular?" },
-  { text: "Blabla" }
-]
-
-function randomQuestion(): Question {
-  const index = Math.floor(Math.random() * questions.length)
-
-  return questions[index]
-}
 
 @Component({
   selector: 'app-board',
   template: `
     <div>
-      board works!
-      <button (click)="loadQuestion()">Nouvelle question</button>
       <div>
-        <app-question [question]="displayedQuestion" [reversed]="true"></app-question>
+      <div *ngIf="game.remainingQuestions.length; else noQuestion">
+        <app-question [question]=game.remainingQuestions[0] ></app-question>
+        <button (click)="loadQuestion()">Next</button>
+        </div>
+        <ng-template #noQuestion>Il n'y a plus de Questions</ng-template>
 
-        <div *ngIf="displayedCriterions.length; else noCriterion">
-          <app-criterion-card *ngFor="let c of displayedCriterions" [criterion]="c" (validated)="onCriterionValidated(c)"></app-criterion-card>
+        <div *ngIf="game.remainingCriterions.length; else noCriterion">
+          <app-criterion-card *ngFor="let c of game.remainingCriterions.slice(0, 3)" [criterion]="c" (validated)="onCriterionValidated(c)"></app-criterion-card>
         </div>
         <ng-template #noCriterion>Tous les critères ont été validés, bravo!</ng-template>
-
-        <app-question [question]="displayedQuestion"></app-question>
       </div>
     </div>
   `,
   styles: [
   ]
 })
+
 export class BoardComponent implements OnInit {
-  displayedQuestion: Question | null = null
-  displayedCriterions = [
-    { text: "Expliquer" },
-    { text: "Expliquer" },
-    { text: "Expliquer" },
-    { text: "Expliquer" },
-    { text: "Decrire" },
-    { text: "Plaisanter" }
-  ]
 
   @Input() game: Game | null = null;
 
   constructor() { }
 
   ngOnInit(): void {
   }
-
+ 
+  //removes criterion from game.remainingCriterion and adds criterion to game.validatedCriterion
+  // What should I do with the error ? 
   onCriterionValidated(c: Criterion) {
-    this.displayedCriterions = this.displayedCriterions.filter(
+    if (this.game ==null) return;
+    this.game.remainingCriterions = this.game.remainingCriterions.filter(
+
       e => e.text != c.text
```

Comment:
This was nice for the first experiment, but here, if a criterion 'A' has been validated, you will want to keep other copies of the same criterion in the `remainingCriterions` array